community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java

```diff
@@ -42,51 +41,96 @@
+            this.internalLogRotationDelay = internalLogRotationDelay;
+            this.maxInternalLogArchives = maxInternalLogArchives;
+            this.rotationExecutor = rotationExecutor;
+            return this;
+        }
+
+        public Builder withRotationListener( Consumer<LogProvider> rotationListener )
+        {
+            this.rotationListener = rotationListener;
+            return this;
+        }
+
+        public StoreLogService inStoreDirectory( FileSystemAbstraction fileSystem, File storeDir ) throws IOException
+        {
+            return toFile( fileSystem, new File( storeDir, INTERNAL_LOG_NAME ) );
+        }
+
+        public StoreLogService toFile( FileSystemAbstraction fileSystem, File internalLogPath ) throws IOException
+        {
+            return new StoreLogService(
+                    userLogProvider,
+                    fileSystem, internalLogPath,
+                    internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives, rotationExecutor, rotationListener );
+        }
+    }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config, JobScheduler jobScheduler ) throws IOException
+    public static Builder withUserLogProvider( LogProvider userLogProvider )
     {
-        this( userLogProvider, fileSystem, storeDir, config, 0L, 0, 0, jobScheduler, Consumers.<LogProvider>noop() );
+        return new Builder().withUserLogProvider( userLogProvider );
     }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config,
-                            long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives,
-                            JobScheduler jobScheduler, final Consumer<LogProvider> rotationListener ) throws IOException
+    public static Builder withRotation( long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives, JobScheduler jobScheduler )
     {
-        this( userLogProvider, fileSystem, storeDir, config, internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives,
-                jobScheduler.executor( JobScheduler.Groups.internalLogRotation ), rotationListener );
+        return new Builder().withRotation( internalLogRotationThreshold, internalLogRotationDelay, maxInternalLogArchives, jobScheduler );
     }
 
-    public StoreLogService( LogProvider userLogProvider, FileSystemAbstraction fileSystem, File storeDir, Config config,
-                            long internalLogRotationThreshold, int internalLogRotationDelay, int maxInternalLogArchives,
-                            Executor rotationExecutor, final Consumer<LogProvider> rotationListener ) throws IOException
+    public static StoreLogService inStoreDirectory( FileSystemAbstraction fileSystem, File storeDir ) throws IOException
```

Comment:
As result in new `StoreLogService` we will have Builder which provides 4 methods that we can use to configure new logService, static factory method `inStoreDirectory` that will provide constructed logService and 2 other static methods that will provide storeLogService builders. Can we simplify that and provide only one way of creating service or builder or factory methods that will return logService?